aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceClient.java

```diff
@@ -961,4 +961,8 @@ protected boolean calculateCRC32FromCompressedData() {
     public String getSignerOverride() {
         return clientConfiguration.getSignerOverride();
     }
+
+    public ClientConfiguration getClientConfiguration() {
+       return clientConfiguration;
```

Comment:
Can we return a copy here?

Reply:
Sure: https://github.com/aws/aws-sdk-java/pull/1641/commits/e85d0dc28024aa7c4ace47cc5beb382d8e452faf